### PR TITLE
CASMCMS-8388 Clarifies BOS v2 uses PCS instead of CAPMC, adds BOSv2 single node troubleshooting

### PR DESCRIPTION
**Pull request by jsl-hpe**
This mod increases the amount of documentation specific to troubleshooting the actions that BOS v2 takes against a specific (or single) node. This gives users parity regarding the operations that have been done for a specific node or nodes in order to view each node's boot progression.

This mod updates language regarding BOSv2's switch to using PCS instead of CAPMC. Existing documentation pieces that called out CAPMC's use were additionally flagged as being BOS v1 specific.

Satisfies JIRA issues:
- [CASMCMS-8388](https://jira-pro.its.hpecorp.net:8443/browse/CASMCMS-8388)
- [CAST-32362](https://jira-pro.its.hpecorp.net:8443/browse/CAST-32362)

# Description

# Checklist

- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

 
[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: operations/boot_orchestration/Cheatsheet.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ cray bos v2 sessions create --template-name SESSION_TEMPLATE_NAME --operation Re
 
 ### Single node commands (v2)
 
-(`ncn-mw#`) Boot a single node:
+c
```
Comment (rustydb) [inline operations/boot_orchestration/Cheatsheet.md:37]: is this intentional?

File: operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md [hunk 2]
```diff
@@ -2,7 +2,48 @@
...
 
-The ConMan tool collects compute node logs. To learn more about ConMan, refer to [ConMan](../conman/ConMan.md).
+BOS v2 implements rich, per-component records for underlying actions that have been applied as part of BOS session
+provisioning. Often, it is helpful to observe the set of operations that BOS has enacted on behalf of a session as they
+apply to a single failing node. The records of operations that have been applied for a node, as well as the intended
+next steps, can be viewed through the BOS v2 component information for the affected hardware.
 
-A node's console data can be accessed through its log file, as described in [Access Compute Node Logs](../conman/Access_Compute_Node_Logs.md)). This information can also be accessed by connecting to the node's console with `ipmitool`. Refer to online documentation to learn more about using ipmitool.
+(`ncn-mw#`)
+```bash
```
Comment (rustydb) [inline operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md:11]: Styling fixes are needed, per style check gate.

**Review by rustydb — CHANGES_REQUESTED** [review]

File: operations/boot_orchestration/Cheatsheet.md (hunk 1 above)
Comment (jsl-hpe) [reply operations/boot_orchestration/Cheatsheet.md:37]: No; apparently I missed a file in the recent merge. Fixing.

File: operations/boot_orchestration/Troubleshoot_Compute_Node_Boot_Issues_Related_to_Slow_Boot_Times.md [hunk 3]
```diff
@@ -1,16 +1,21 @@
 # Troubleshoot Compute Node Boot Issues Related to Slow Boot Times
 
-Inspect BOS, the Boot Orchestration Agent \(BOA\) job logs, and the Configuration Framework Service \(CFS\) job logs to obtain information that is critical for boot troubleshooting. Use this procedure to determine why compute nodes are booting slower than expected.
+> **`NOTE`** This section is for BOS V1 only.  BOS V2 does not use CAPMC, nor does it have BOA. However, the steps
+> outlined below allow for similar debugging slow steps within CFS initiated sessions.
```
Comment (jsollom-hpe) [inline operations/boot_orchestration/Troubleshoot_Compute_Node_Boot_Issues_Related_to_Slow_Boot_Times.md:4]: debugging 'of' slows steps

**Review by jsollom-hpe — CHANGES_REQUESTED** [review]

File: operations/boot_orchestration/Cheatsheet.md [hunk 4]
```diff
@@ -37,7 +37,7 @@ cray bos v2 sessions create --template-name SESSION_TEMPLATE_NAME --operation Re
 (`ncn-mw#`) Boot a single node:
 
 ```bash
-cray bos v2 sessions create --template-name SESSION_TEMPLATE_NAME --operation Boot --limit <node's xname>
+cray bos v1 session create --template-name SESSION_TEMPLATE_NAME --operation Boot --limit <node's xname>
```
[the remark this answers is not in the document]
Comment (jsl-hpe) [reply operations/boot_orchestration/Cheatsheet.md:40]: Updated.

File: operations/boot_orchestration/Limitations_for_Gigabyte_BMC_Hardware.md [hunk 5]
```diff
@@ -1,5 +1,7 @@
 # BOS Limitations for Gigabyte BMC Hardware
 
+> **`NOTE`** This section is for BOS V1 only.  BOS V2 does not use CAPMC.
```
[the remark this answers is not in the document]
Comment (jsl-hpe) [reply operations/boot_orchestration/Limitations_for_Gigabyte_BMC_Hardware.md:3]: That's correct; BOS v1 will be deprecated before CAPMC goes away. Though in truth, they made CAPMC a shim to PCS anyway, so everything eventually talks PCS.

File: operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md [hunk 6]
```diff
@@ -1,8 +1,52 @@
 # Node Boot Root Cause Analysis
 
-The first step in debugging compute node boot-related issues is to determine the underlying cause, and the stage that the issue was encountered at.
+The first step in debugging compute node boot-related issues is to determine the underlying cause, and the stage that
+the issue was encountered at.
 
-The ConMan tool collects compute node logs. To learn more about ConMan, refer to [ConMan](../conman/ConMan.md).
+BOS v2 implements rich, per-component records for underlying actions that have been applied as part of BOS session
```
[the remark this answers is not in the document]
Comment (jsl-hpe) [reply operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md:6]: Updated.

File: operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md [hunk 7]
```diff
@@ -1,8 +1,52 @@
 # Node Boot Root Cause Analysis
 
-The first step in debugging compute node boot-related issues is to determine the underlying cause, and the stage that the issue was encountered at.
+The first step in debugging compute node boot-related issues is to determine the underlying cause, and the stage that
```
[the remark this answers is not in the document]
Comment (jsl-hpe) [reply operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md:3]: Done.

File: operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md [hunk 8]
```diff
@@ -1,8 +1,52 @@
...
+the issue was encountered at.
 
-The ConMan tool collects compute node logs. To learn more about ConMan, refer to [ConMan](../conman/ConMan.md).
+BOS v2 implements rich, per-component records for underlying actions that have been applied as part of BOS session
+provisioning. Often, it is helpful to observe the set of operations that BOS has enacted on behalf of a session as they
```
[the remark this answers is not in the document]
Comment (jsl-hpe) [reply operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md:7]: I added an additional sentence with this suggestion.

File: operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md [hunk 9]
```diff
@@ -1,8 +1,52 @@
...
+to the node's console with `ipmitool`. Refer to online documentation to learn more about using `ipmitool`.
+
+If the node has booted into a multi-user target phase, but BOS has not completed booting the node, the node may have
+encountered a configuration error. A similar set of records for configuration for a given node can be obtained from
+[CFS](../configuration_management) endpoint for the same hardware component.
+
+(`ncn-mw#`) Verify the configuration status of a CFS component of the same name
+
+```bash
+cray cfs components describe x3000c0s1b0n0
```
[the remark this answers is not in the document]
Comment (jsl-hpe) [reply operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md:37]: See existing formatting from cheatsheet.md.

[the remark this answers is not in the document]
Comment (jsollom-hpe) [reply operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md:37]: Noted. I was wrong.

File: operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md [hunk 10]
```diff
@@ -1,8 +1,52 @@
...
+
+If a node has been booted with BOS as part of a boot or reboot operation, and the node was powered on, but has not
+begun configuring, the node may be stuck in early initialization (Failure to `iPXE` chain, network setup issues, failure to
+obtain a root filesystem, or other dracut module specific issues). In this case, it is best to connect to the node's
+console logs to obtain specific information about the failed node. To learn more about ConMan, refer to
+[ConMan](../conman/ConMan.md). A node's console data can be accessed through its log file, as described in
+[Access Compute Node Logs](../conman/Access_Compute_Node_Logs.md)). This information can also be accessed by connecting
+to the node's console with `ipmitool`. Refer to online documentation to learn more about using `ipmitool`.
+
+If the node has booted into a multi-user target phase, but BOS has not completed booting the node, the node may have
```
[the remark this answers is not in the document]
Comment (jsl-hpe) [reply operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md:30]: I've updated this last paragraph to indicate that A BOSv2 configuring status may mean to check on the CFS component records.

File: operations/boot_orchestration/Troubleshoot_Compute_Node_Boot_Issues_Related_to_Slow_Boot_Times.md (hunk 3 above)
Comment (jsl-hpe) [reply operations/boot_orchestration/Troubleshoot_Compute_Node_Boot_Issues_Related_to_Slow_Boot_Times.md:4]: ammended.

File: operations/boot_orchestration/Cheatsheet.md [hunk 11]
```diff
@@ -37,7 +37,7 @@ cray bos v2 sessions create --template-name SESSION_TEMPLATE_NAME --operation Re
 (`ncn-mw#`) Boot a single node:
 
 ```bash
-cray bos v2 sessions create --template-name SESSION_TEMPLATE_NAME --operation Boot --limit <node's xname>
+cray bos v2 session create --template-name SESSION_TEMPLATE_NAME --operation Boot --limit <node's xname>
```
Comment (rbak-hpe) [inline operations/boot_orchestration/Cheatsheet.md:40]: I'm pretty sure this isn't correct. v2 uses plural, v1 uses singular.

**Review by rbak-hpe — APPROVED** [review]

File: operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md [hunk 12]
```diff
@@ -1,8 +1,56 @@
...
+encountered a configuration error. A similar set of records for configuration for a given node can be obtained from
+[CFS](../configuration_management) endpoint for the same hardware component. In this case, BOS will indicate the
+component status is `configuring`, and further querying information from CFS for the same component may be in order.
+
+(`ncn-mw#`) Verify the configuration status of a CFS component of the same name
+
+```bash
+cray cfs components describe x3000c0s1b0n0
```
Comment (rustydb) [inline operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md:41]: nit: This looks like a command and output, please break this up into "command" and "potential output" for easier copy-paste.

Along with the other similar lines such as line 13.

File: operations/boot_orchestration/BOS_API_Versions.md [hunk 13]
```diff
@@ -21,6 +21,7 @@ There are some different API responses in BOS v2 as well:
 
 - Listing sessions in BOS v2 results in full session records rather than just a list of the current session names.
 - The session status endpoint has also changed significantly. See [View the Status of a BOS Session](View_the_Status_of_a_BOS_Session.md) for more information on the session status endpoint.
+- BOS v2 leverages the Power Control Service (PCS)
```
Comment (nrockershousen) [inline operations/boot_orchestration/BOS_API_Versions.md:24]:
```suggestion
- BOS v2 leverages the Power Control Service (PCS).
```

File: operations/boot_orchestration/Boot_Orchestration.md [hunk 14]
```diff
@@ -14,7 +14,7 @@ BOS depends on each of the following services to complete its tasks:
 
 - Boot Script Service \(BSS\) - Stores the configuration information that is used to boot each hardware component. Nodes consult BSS for their boot artifacts and boot parameters when nodes boot or reboot.
 - Configuration Framework Service \(CFS\) - BOA launches CFS to apply configuration to the nodes in its boot sets \(node personalization\).
-- Cray Advanced Platform Monitoring and Control \(CAPMC\) - Used to power on and off the nodes.
+- Power Control Service \(PCS\) - Used to power nodes on and off, as well as query current power status
```
Comment (nrockershousen) [inline operations/boot_orchestration/Boot_Orchestration.md:17]:
```suggestion
- Power Control Service \(PCS\) - Used to power nodes on and off, as well as query current power status.
```

File: operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md [hunk 15]
```diff
@@ -1,8 +1,56 @@
...
+the issue was encountered at.
 
-The ConMan tool collects compute node logs. To learn more about ConMan, refer to [ConMan](../conman/ConMan.md).
+BOS v2 provides rich, per-component records for underlying actions that have been applied as part of BOS session
+provisioning. Often, it is helpful to observe the set of operations that BOS has enacted on behalf of a session as they
+apply to a single failing node. The records of operations that have been applied for a node, as well as the intended
+next steps, can be viewed through the BOS v2 component information for the affected hardware.
 
-A node's console data can be accessed through its log file, as described in [Access Compute Node Logs](../conman/Access_Compute_Node_Logs.md)). This information can also be accessed by connecting to the node's console with `ipmitool`. Refer to online documentation to learn more about using ipmitool.
+(`ncn-mw#`) Verify the status of a BOS component
```
Comment (nrockershousen) [inline operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md:11]:
```suggestion
(`ncn-mw#`) Verify the status of a BOS component.
```

File: operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md [hunk 16]
```diff
@@ -1,8 +1,56 @@
...
+[ConMan](../conman/ConMan.md). A node's console data can be accessed through its log file, as described in
+[Access Compute Node Logs](../conman/Access_Compute_Node_Logs.md)). This information can also be accessed by connecting
+to the node's console with `ipmitool`. Refer to online documentation to learn more about using `ipmitool`.
+
+If the node has booted into a multi-user target phase, but BOS has not completed booting the node, the node may have
+encountered a configuration error. A similar set of records for configuration for a given node can be obtained from
+[CFS](../configuration_management) endpoint for the same hardware component. In this case, BOS will indicate the
+component status is `configuring`, and further querying information from CFS for the same component may be in order.
+
+(`ncn-mw#`) Verify the configuration status of a CFS component of the same name
```
Comment (nrockershousen) [inline operations/boot_orchestration/Node_Boot_Root_Cause_Analysis.md:38]:
```suggestion
(`ncn-mw#`) Verify the configuration status of a CFS component of the same name.
```

**Comment by rustydb** [comment]
After the review comments are addressed, does this need a backport to 1.4 or 1.3? The JIRA had no fix version, I added CSM V1.5.0 since this PR was targeting main.

**Review by jsollom-hpe — APPROVED** [review]

**Comment by jsl-hpe** [comment]
A new JIRA task to standardize all boot orchestration commands and formatting to prevent future gotchas while maintaining the doc: https://jira-pro.its.hpecorp.net:8443/browse/CASMCMS-8466

**Pull request merged**